### ISSUE TITLE
calling ConfigService with default value causes ts(2589) - Type instantiation is excessively deep and possibly infinite

### ISSUE THREAD

**Issue by robertwho** [nestjs/config#625]
## I'm submitting a...

<pre><code>
[x] Regression 
[ ] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

The compile time error
```
const config = new ConfigService(...)
config.get('A_KEY', 'default')

>> Type instantiation is excessively deep and possibly infinite. ts(2589)
```

is shown, when using config.get() with a default value.

## Expected behavior

Above code snippet should compile without any complains.

## Minimal reproduction of the problem with instructions

[Playground Example](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgMIHsQ2AcwOIRgDyADmMJgM7IDeAUMsqDNAFzJhQCuEA3HQF86dMAE8SKAHLoAklmgAVcRAA8CgHzIAvMgDaCgLr7kEAB6QQAE2pwQo5AH5kABmTsQEAG7QD-EcuQABTgwAAsZAFsSABs1ABpkAGkIezMLa2QAaxT0GGQNbSSUk3MIK2pKTlAcBkd83WTRAxL06gAlCAR0KEsVSqhqhNtRdVrGBzHGKYAfZAADABIaRoEAOiXgsMiYtQaUgwSAUVMEaK5LVWzRXPrGg6ycvOHdA3VNADJJqe-kfuqBOZfGbzJYrdY0Y6nc6XR63fYJK43Z6vZDvX5VEA4AGTdxeaC1XHeKB+MQSIIhcJRaIAJjUmh0m0pOwUCNhBVmiLyChJAUZdMKjO2NP5aTKGT+mOQHLZtScgqptI0BIe1y5PLJjIAanAzqpaizaoEWmLqHylfSgsbyiCaMxoEVRGslnaoMgOpVseMHVaMpz8pMnO6wD7TRTdndRj86lqdTxw-C3RBKpGfoT8d801BlUbRda-QpZfVAgZlR4iX5TnBKNQMFhcABlaCeYBIFSJQodLo9PoYnBDOxvWjCRg4QhqQqJBI50rWs0W4YJNoCina3XxIJvAAUkxIUHQEigYkZ7ECcUm+-IVHYtew+EIpEvIEoZ8YAEp2EvZlwrBBsB5LPwI5jgohTDOom67vu0BHhS7B+pOyAXPAXDRGAq48Ow0hyCwUBKBIdLvvkgHIKOYDjjoYHbt8SFwChaGxhARBQA+FBPg47AgbMN64AQxBkKxz61IRnHIN+SGgBAlhDpMUCEFwUAgIhv60ah6GMcx-FUMgVZEbUQjCPpXRPsGRm3oUHgAO5oJgt6NlAzZIJur7CEZlDoNEECrNE6A4Jupm4KspGbgA5JIACqACyABChxtAA+pqACCAAyYWHMFCTBTRdHBa+zmMEAA)

## What is the motivation / use case for changing the behavior?

nestjs/config v1.0.0 should work as v0.6.3 did

## Environment

<pre><code>
Nest version: 8.0.4
Nest config version: 1.0.0
</code></pre>

**Comment by micalevisk** [comment]
you could use `config.get<string>('NUMBER_VALUE', 'default')` as a temp fix.

**Comment by robertwho** [comment]
Many thanks! Wasn't aware of using the type parameter after the get method.

**Comment by kamilmysliwiec** [comment]
Let's keep it open since the original syntax you used should work properly as well

**Comment by kamilmysliwiec** [comment]
Fixed in 1.0.1

**Comment by micalevisk** [comment]
@M-Hoogie yeah. I just tested with `@nestjs/config@1.1.0` (the latest)

```ts
const config = app.get(ConfigService);
//    ^ ConfigService<any, any>
config.get('A_KEY', 'default') // TSC error ❌

// --------------------------------------------------------- //

const config = app.get<ConfigService>(ConfigService);
//    ^ ConfigService<Record<string, unknown>, false>
config.get('A_KEY', 'default') // ✅
```

any thoughts on this @kamilmysliwiec?

**Comment by micalevisk** [comment]
fixed by commit b85f734
